### PR TITLE
Adding the inference microservice

### PR DESCRIPTION
**Pull request by cdettmering**
Added FastAPI to current feature list

### CODE REVIEW

File: src/anyGPT/service/app.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# Runner dependency
+def get_runner():
+    return AnyGPTRunner(model)
+
+
+# Runs inference on the model
+@app.post("/infer")
+async def infer(request: InferenceRequest, runner: AnyGPTRunner = Depends(get_runner)):
+    print(f"Running inference on {request.data}")
+    return runner.sample(request.data)
```
Comment (miguelalonsojr) [inline src/anyGPT/service/app.py]: can you add the additional parameters available to the inputs of the post request? it's max new tokens, temperature, and top k.

**Review by miguelalonsojr — CHANGES_REQUESTED** [review]

File: src/anyGPT/service/app.py (hunk 1 above)
Comment (cdettmering-unity) [reply src/anyGPT/service/app.py]: This is fixed

File: .gitignore [hunk 2]
```diff
@@ -163,3 +163,5 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
+data/
```
Comment (miguelalonsojr) [inline .gitignore]: Can you change this to checkpoints instead of data? I want to avoid collisions with the data module in the package.

**Review by miguelalonsojr — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (cdettmering) [reply .gitignore]: Fixed

**Review by miguelalonsojr — APPROVED** [review]
LGTM

**Pull request merged**